assets/js/nav.js

```diff
@@ -1,61 +1,31 @@
 $(document).ready(function () {
-    $("#mobile-navigation").click(function (e) {
-        var target = e.target;
-        var classes = $(target).attr('class');
-        var toggleClass = "visible";
-        var root = $("#mobile-navigation");
-        var mainMenuWrap = $(".main-menu-container");
+    var activeClass = 'mobile-menu-active';
+    var slideTo = function(element){
         var mainMenu = $(".main-menu");
-        var currentMenu = $(".visible");
 
-        /*
-            traverse up and down the menu:
-            remove visible-menu-block from current menu and
-            clone target menu and append to root of mobile navigation structure
-        */
-        if ((classes !== undefined) && (classes.indexOf("has-children") !== -1)) {
-            var targetMenu = $(target).children("ul");
+        element.children('ul').show();
+        element.siblings().children('ul').hide();
+        $('.' + activeClass).removeClass(activeClass);
 
-            if (isMainMenu()) {
-                $(currentMenu).toggleClass(toggleClass);
-            } else {
-                $(currentMenu).remove();
-            }
+        var level = element.parents('ul').length;
+        mainMenu.css('left', (level * -100) + '%');
+        element.addClass(activeClass);
 
-            $(mainMenuWrap).append(targetMenu.clone().toggleClass(toggleClass));
+        $(".menu-icon-navigation").attr('src', (level === 0) ? '/assets/img/mobile-menu-hamburger-icon.png' : '/assets/img/mobile-menu-back-icon.png');
+    };
 
-        } else if ((classes !== undefined) && (classes.indexOf("menu-icon-navigation") !== -1)) {
-            var targetClass = $(".visible").attr('class').replace("visible","").trim();
-            var targetMenu = $(".main-menu").find("ul." + targetClass).parent().parent();
+    $('.main-menu').click(function (e) {
+        var target = $(e.target);
 
-            if ($(".main-menu.visible").length !== 0) {
-                return;
-            } else if (targetMenu.attr('class') === "main-menu") {
-                $(currentMenu).remove();
-                $(mainMenu).toggleClass(toggleClass);
-            } else {
-                $(currentMenu).remove();
-                $(mainMenuWrap).append(targetMenu.clone().toggleClass(toggleClass));
-            }
-
-        } else {
-            target.click();
-        }
-
-        function isMainMenu () {
-            return ($(".main-menu.visible").length) ? true : false
-        }
-
-        function menuIconBackNavigation () {
-            $(".menu-icon-navigation").attr("src", "/assets/img/mobile-menu-back-icon.png");
+        if (target.hasClass('has-children')) {
```

Comment:
Eli said that class is being used for CSS, so it makes sense to keep it this way.